### PR TITLE
check the consistance of nullifier and note_commitment

### LINKED ISSUE #120: Check consistency of nf and cm for VPs in PartialTransaction::verify
We must guarantee that all VPs in the partial transaction take the same notes from the action circuits as inputs.

We can explicitly check the consistency of nf for spend_note and cm for output_note since nf and cm are public inputs in the circuits.

### PR DESCRIPTION
**Pull request by XuyangSong**
To fix #120

### CODE REVIEW

File: taiga_halo2/src/circuit/vp_circuit.rs [hunk 1]
```diff
@@ -32,6 +32,14 @@ impl VPVerifyingInfo {
         let params = SETUP_PARAMS_MAP.get(&VP_CIRCUIT_PARAMS_SIZE).unwrap();
         self.proof.verify(&self.vk, params, &[&self.instance])
     }
+
+    pub fn get_nullifiers(&self) -> [pallas::Base; NUM_NOTE] {
+        [self.instance[0], self.instance[2]]
```
Comment (Acentelles) [inline taiga_halo2/src/circuit/vp_circuit.rs:37]: It looks a bit clunky. What's the logic behind this indexes? Would it be possible to make it more declarative? Can it go wrong (e.g. by instances not having at least for elements or by having elements in the wrong position)?

File: taiga_halo2/src/transaction.rs [hunk 2]
```diff
@@ -125,14 +134,14 @@ impl Transaction {
         self.partial_txs.push(ptx);
     }
 
-    pub fn get_nullifiers(&self) -> Vec<Nullifier> {
+    pub fn get_nullifiers(&self) -> Vec<pallas::Base> {
```
Comment (Acentelles) [inline taiga_halo2/src/transaction.rs:137]: Would it be possible to keep the types of Nullifier and NoteCommitment?

**Review by Acentelles — APPROVED** [review]
Looks good to me mod some comments

File: taiga_halo2/src/circuit/vp_circuit.rs (hunk 1 above)
Comment (XuyangSong) [reply taiga_halo2/src/circuit/vp_circuit.rs:37]: I'll add the constant index names to make it more declarative

**Pull request merged**